chainerio/filesystems/hdfs.py

```diff
@@ -71,6 +50,21 @@ def _dump_read_file(self, filepath, content):
             dump_file.write(content)
         return
 
+    def _fileobject_returner(self, file_obj: Type['IOBase'],
+                             file_path: str, mode: str = 'rb',
+                             buffering: int = -1,
+                             encoding: Optional[str] = None,
+                             errors: Optional[str] = None,
+                             newline: Optional[str] = None,
+                             closefd: bool = True,
+                             opener: Optional[Callable[
+                                 [str, int], Any]] = None) -> Type['IOBase']:
+
+        if 'b' not in mode:
+            file_obj = io.TextIOWrapper(file_obj, encoding, errors, newline)
+
```

Comment:
Do you work on the issue that PyArrow's file object doesn't have `peek` method here? Or another PR?

Reply:
I would like to do it in another PR since the problem is different

Reply:
The `peek` in HDFS is different from POSIX where the peek was there. For HDFS, we need either add a `peek` method to HDFS file object or create our own pickle.